### PR TITLE
[RISCV] Fold vfmv.f.s into load from stack

### PR DESCRIPTION
**Pull request by lukel97**
This is the f64/f32 version of #109774.
I've left out f16 and bf16 for now because there's a separate issue where we can't select extract_vector_elt when f16/bf16 is a legal type, see #110126.

### CODE REVIEW

File: llvm/test/CodeGen/RISCV/rvv/stack-folding.ll [hunk 1]
```diff
@@ -160,3 +160,103 @@ truebb:
...
+; RV32-NEXT:    vs1r.v v8, (a1) # Unknown-size Folded Spill
+; RV32-NEXT:    andi a0, a0, 1
+; RV32-NEXT:    #APP
+; RV32-NEXT:    #NO_APP
+; RV32-NEXT:    beqz a0, .LBB4_2
+; RV32-NEXT:  # %bb.1: # %truebb
+; RV32-NEXT:    addi a0, sp, 16
+; RV32-NEXT:    vl1r.v v8, (a0) # Unknown-size Folded Reload
+; RV32-NEXT:    vsetivli zero, 1, e64, m1, ta, ma
+; RV32-NEXT:    vfmv.f.s fa0, v8
```
Comment (topperc) [inline llvm/test/CodeGen/RISCV/rvv/stack-folding.ll:183]: Why didn't it work for RV32?

File: llvm/lib/Target/RISCV/RISCVInstrInfo.cpp [hunk 2]
```diff
@@ -784,6 +784,26 @@ MachineInstr *RISCVInstrInfo::foldMemoryOperandImpl(
       }
       break;
     }
+    if (RISCV::getRVVMCOpcode(MI.getOpcode()) == RISCV::VFMV_F_S) {
+      unsigned Log2SEW =
+          MI.getOperand(RISCVII::getSEWOpNum(MI.getDesc())).getImm();
+      if (STI.getXLen() < (1 << Log2SEW))
```
Comment (topperc) [inline llvm/lib/Target/RISCV/RISCVInstrInfo.cpp:790]: Is this XLen checked needed?

File: llvm/test/CodeGen/RISCV/rvv/stack-folding.ll (hunk 1 above)
Comment (topperc) [reply llvm/test/CodeGen/RISCV/rvv/stack-folding.ll:183]: I guess it's the XLen check?

File: llvm/lib/Target/RISCV/RISCVInstrInfo.cpp (hunk 2 above)
Comment (lukel97) [reply llvm/lib/Target/RISCV/RISCVInstrInfo.cpp]: Woops it's not

**Review by topperc — APPROVED** [review]
LGTM

**Review by preames — APPROVED** [review]
LGTM

**Pull request merged**